From: Bhushan Shah <bhushan.shah@machinesoul.in>
Date: Tue, 11 Nov 2025 19:48:46 +0530
Subject: Revert "FROMGIT ASoC: qcom: sm8250: explicitly set format in
 sm8250_be_hw_params_fixup()"

The AW88261 amplifier on Fairphone 5 does not initialize with the format
limited to S16_LE.

This reverts commit 43a348410de5220d30ae0dcd432c4fed50bc572f.
---
 sound/soc/qcom/sm8250.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/sound/soc/qcom/sm8250.c b/sound/soc/qcom/sm8250.c
index 3feecfc..20e6f93 100644
--- a/sound/soc/qcom/sm8250.c
+++ b/sound/soc/qcom/sm8250.c
@@ -7,7 +7,6 @@
 #include <sound/soc.h>
 #include <sound/soc-dapm.h>
 #include <sound/pcm.h>
-#include <sound/pcm_params.h>
 #include <linux/soundwire/sdw.h>
 #include <sound/jack.h>
 #include <linux/input-event-codes.h>
@@ -62,11 +61,9 @@ static int sm8250_be_hw_params_fixup(struct snd_soc_pcm_runtime *rtd,
 					SNDRV_PCM_HW_PARAM_RATE);
 	struct snd_interval *channels = hw_param_interval(params,
 					SNDRV_PCM_HW_PARAM_CHANNELS);
-	struct snd_mask *fmt = hw_param_mask(params, SNDRV_PCM_HW_PARAM_FORMAT);
 
 	rate->min = rate->max = 48000;
 	channels->min = channels->max = 2;
-	snd_mask_set_format(fmt, SNDRV_PCM_FORMAT_S16_LE);
 
 	return 0;
 }
